run.py script fails on linux due to shell=True

# Issue:
The mvn subprocess for "Initializing Maven project in: "hytale-server" fails on linux due to parameter `shell=True`. `shell=True` will only run the first thing in the list which is mvn, but doesn't include the rest of the list in the command. It's possible this fails on other OSs as well, but I only have linux, so I can't test on other OSs.

# Solution:
Remove the `shell=True` parameter from the subprocess.run function args, or use single string for command. (I implemented the former to get it to work).

# Error:
```
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.030 s
[INFO] Finished at: 2026-01-16T18:13:05-05:00
[INFO] ------------------------------------------------------------------------
[ERROR] No goals have been specified for this build. You must specify a valid lifecycle phase or a goal in the format <plugin-prefix>:<goal> or <plugin-group-id>:<plugin-artifact-id>[:<plugin-version>]:<goal>. Available lifecycle phases are: pre-clean, clean, post-clean, validate, initialize, generate-sources, process-sources, generate-resources, process-resources, compile, process-classes, generate-test-sources, process-test-sources, generate-test-resources, process-test-resources, test-compile, process-test-classes, test, prepare-package, package, pre-integration-test, integration-test, post-integration-test, verify, install, deploy, pre-site, site, post-site, site-deploy. -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/NoGoalSpecifiedException
Traceback (most recent call last):
  File "/home/qube/Programming/Home/HytaleMods/patcher/run.py", line 193, in <module>
    subprocess.run([
    ~~~~~~~~~~~~~~^^
        "mvn", "archetype:generate",
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<2 lines>...
        "-DarchetypeArtifactId=maven-archetype-quickstart", "-DinteractiveMode=false"
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ], check=True, shell=True)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.14/subprocess.py", line 577, in run
    raise CalledProcessError(retcode, process.args,
                             output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['mvn', 'archetype:generate', '-DgroupId=com.hypixel.hytale', '-DartifactId=hytale-server', '-DarchetypeArtifactId=maven-archetype-quickstart', '-DinteractiveMode=false']' returned non-zero exit status 1.
```

--- a/run.py
+++ b/run.py
@@ -190,12 +190,17 @@ if __name__ == "__main__":
             # mvn archetype:generate -DgroupId=com.hypixel.hytale -DartifactId=hytale-server -DarchetypeArtifactId=maven‑archetype‑quickstart -DinteractiveMode=false
             logger.info("\n\nInitializing Maven project in:\n{}\n\n", Constants.PROJECT_DIR)
 
+            # Use shell if on windows else do not see:
+            # https://github.com/HytaleModding/patcher/issues/5
+            # https://github.com/HytaleModding/patcher/issues/9
+            use_shell = os.name == 'nt' 
+
             subprocess.run([
                 "mvn", "archetype:generate",
                 # "-DgroupId=com.hypixel.hytale", "-DartifactId=hytale-server",
                 "-DgroupId=com.hypixel.hytale", "-DartifactId=" + Constants.PROJECT_DIR.name,
                 "-DarchetypeArtifactId=maven-archetype-quickstart", "-DinteractiveMode=false"
-            ], check=True, shell=True)
+            ], check=True, shell=use_shell)
 
             logger.info("Maven project initialized!")
 

